fix: add recurringEventCount to URL params

ibex088 opened the pull request:
## What does this PR do?

- Fixes #18547

#### Video Demo (if applicable):

https://www.loom.com/share/78d5d072b3234197886d94d3c514c53e



















<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Validate the occurrence count from the ?recurringEventCount= URL param and unify input validation for recurring events. Invalid values now fall back to the allowed maximum, show a warning, and the store stays in sync with the URL when allowed. Adds a “Repeats X times” badge when count > 1.

- **Bug Fixes**
  - Initialize occurrence count from the recurringEventCount query param on load; validate numeric, >=1, and <= max occurrences, and default to max when missing.
  - Guard against invalid/NaN values updating state or URL.
  - Prevent overlayCalendar toggle from overwriting existing URL query params.

- **New Features**
  - Show a “Repeats {{count}} times” badge in the booking modal (new i18n key: repeats_num_times).

<sup>Written for commit 3be6bb63b043fd830ae0ed305975562750308127. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

ibex088 committed:
fix: validate occurrence count from URL params and refactor validation logic
- Initialize occurrence count from URL query parameter on page load

ibex088 committed:
fix: change query parameter name from occurenceCount to recurringEventCount

supalarry reviewed (commented):
Review:

ibex088 commented on packages/features/bookings/Booker/store.ts:
```diff
@@ -421,8 +426,13 @@ export const createBookerStore = () =>
     },
     recurringEventCount: null,
```
Let me clarify the distinction between these two variables:

## Two-State Pattern for Input Validation

We use a **two-state pattern** to separate UI state from business logic state:

### `occurenceCount` (UI/Input State)
- Stores the **raw user input** (can be invalid)
- Initialized from URL query param `?recurringEventCount=X`
- Always reflects what the user typed or what came from the URL
- Used as the `defaultValue` for the input field
- Gets persisted back to URL params

### `recurringEventCount` (Validated/Business Logic State)
- Stores the **validated value** actually used for creating bookings
- Only gets set when the input is valid (within range 1 to max)
- Used in the booking submission and for displaying recurring dates preview
- Never contains invalid values

## Why This Pattern?

This separation allows us to:

1. **Preserve user input**: If someone types "999" but max is "12", we keep "999" in the input field (with a warning) rather than replacing it immediately
2. **Safe booking flow**: The booking always uses a valid value from `recurringEventCount`, even if `occurenceCount` is invalid
3. **URL param handling**: When someone shares a URL with `?occurenceCount=999`, we validate it and either use it (if valid) or fallback to the max allowed value, while still showing the warning

## Flow Example

User types "999" (max is 12):

occurenceCount = 999 (stored, shown in input)
recurringEventCount = 12 (fallback to max)
warning = true (shown to user)
Booking uses: 12 ✅
User types "5" (max is 12):

occurenceCount = 5 (stored, shown in input)
recurringEventCount = 5 (valid, used for booking)
warning = false
Booking uses: 5 ✅

ibex088 replied:
how does recurringCountInput sound?

supalarry replied:
I think `QueryParam` suffix makes it more clear where it comes from and goes to - and input in my mind could be an input to react component.

ibex088 committed:
refactor: simplify occurrence count validation logic

ibex088 committed:
fix: prevent NaN from being set in recurring event count query parameter

ibex088 committed:
fix: prevent overlay calendar toggle from overwriting query params

ibex088 committed:
fix: sync occurrence count state with max occurrences limit

ibex088 committed:
refactor: rename occurenceCount to recurringEventCountQueryParam

ibex088 committed:
Merge branch 'main' into feat/validate-occurrence-count-from-url

ibex088 committed:
fix: prevent invalid recurring event count from updating state

- Added validation guard to only update state when recurringEventCountQueryParam is valid (not null or NaN)
- Simplified URL update logic by removing unnecessary empty string fallback

ibex088 committed:
feat: add recurring event count badge for mobile (#24991)

* feat: display recurring event count badge in mobile

* feat: update recurring event translation key for clarity

ibex088 committed:
Merge branch 'main' into feat/validate-occurrence-count-from-url

ibex088 committed:
Merge branch 'main' into feat/validate-occurrence-count-from-url
